Bio/AlignIO/MafIO.py

```diff
@@ -0,0 +1,581 @@
+                        "strand": strand,
+                        "srcSize": int(line_split[5])}
+
+                sequence = line_split[6]
+
+                # interpret a dot/period to mean same the first sequence
+                if "." in sequence:
+                    if not records:
+                        raise ValueError("Found dot/period in first sequence of alignment")
+
+                    ref = str(records[0].seq)
+                    new = []
+
+                    for (s, r) in zip(sequence, ref):
+                        new.append(r if s == "." else s)
+
+                    sequence = "".join(new)
+
+                records.append(SeqRecord(Seq(sequence, alphabet),
+                               id=line_split[1],
+                               name=line_split[1],
+                               description="",
+                               annotations=anno))
+            elif line.startswith("e") or \
+                 line.startswith("i") or \
+                 line.startswith("q"):
+                # not implemented
+                pass
+            elif not line.strip():
+                # end a bundle of records
+                if seq_count is not None:
+                    assert len(records) == seq_count
+
+                alignment = MultipleSeqAlignment(records, alphabet)
+                # TODO - Introduce an annotated alignment class?
+                # See also Bio/AlignIO/FastaIO.py for same requirement.
+                # For now, store the annotation a new private property:
+                alignment._annotations = annotations
+
+                yield alignment
+
+                in_a_bundle = False
+
+                annotations = []
+                records = []
+            else:
+                raise ValueError("Error parsing alignment - unexpected line:\n%s" % (line,))
+        elif line.startswith("a"):
+            # start a bundle of records
+            in_a_bundle = True
+
+            if len(line.strip().split()[1:]) != line.count("="):
+                raise ValueError("Error parsing alignment - invalid key in 'a' line")
+
+            annotations = dict([x.split("=") for x in line.strip().split()[1:]])
+        elif line.startswith("#"):
+            # ignore comments
+            pass
+        elif not line:
+            break
+
+
+class MafIndex(object):
+    def __init__(self, sqlite_file, maf_file, target_seqname):
+        """Indexes or loads the index of a MAF file"""
+
+        import os
+
+        try:
+            from sqlite3 import dbapi2 as _sqlite
+        except ImportError:
+            from Bio import MissingPythonDependencyError
+            raise MissingPythonDependencyError("Requires sqlite3, which is "
+                                               "included Python 2.5+")
+
+        self._target_seqname = target_seqname
+        self._maf_file = maf_file
+
+        # make sure maf_file exists, then open it up
+        if os.path.isfile(self._maf_file):
```

Comment:
In this case I am OK with passing any exception straight back to the user if the file does exist but cannot be opened (e.g. due to file permissions). This also avoids the question of if raising a ``ValueError`` is the best choice.